ext/uri/uri_parser_rfc3986.c

```diff
@@ -99,6 +99,21 @@ ZEND_ATTRIBUTE_NONNULL static UriUriA *get_uri_for_reading(php_uri_parser_rfc398
 	}
 }
 
+ZEND_ATTRIBUTE_NONNULL static UriUriA *get_uri_for_writing(uri_internal_t *internal_uri)
+{
+	php_uri_parser_rfc3986_uris *uriparser_uris = internal_uri->uri;
+
+	return &uriparser_uris->uri;
+}
+
+ZEND_ATTRIBUTE_NONNULL static void reset_normalized_uri_after_writing(uri_internal_t *internal_uri)
```

Comment:
Having a separate method for this is risky (it's easy to forget to call it). I would suggest one of two options:

- Do not clone the normalized URI, as I did in #19588: This is likely the most performant solution when expecting that most URIs are modified after being cloned.
- Reset the normalized URI in `get_uri_for_writing()`: We expect the modification to succeed in most cases, in the worst case we're needlessly invalidating the normalized URI and it will just get renormalized.

Both options would reliably fix the issue and leave little room for human error in the implementation.